api/context/rest.go

```diff
@@ -0,0 +1,338 @@
+
+		if resp.NextPageToken == nil {
+			break
+		}
+
+		params.PageToken = resp.NextPageToken
+	}
+	return contexts, nil
+}
+
+type CreateContextWithRestParams struct {
+	Name  string `json:"name"`
+	Owner struct {
+		Type string  `json:"type"`
+		Id   *string `json:"id,omitempty"`
+		Slug *string `json:"slug,omitempty"`
+	} `json:"owner"`
+}
+
+var orgOwnerType = "organization"
+
+// Creates a new context.
+// Binding to https://circleci.com/docs/api/v2/index.html#operation/createContext
+func CreateContextWithRest(c *rest.Client, params *CreateContextWithRestParams) (*Context, error) {
+	if params == nil {
+		return nil, fmt.Errorf("parameters are nil when creating context")
+	}
+	if params.Owner.Id == nil && params.Owner.Slug == nil {
+		return nil, fmt.Errorf("to create a context, need either org ID or org slug, received none")
+	}
+	if params.Owner.Type == "" {
+		params.Owner.Type = orgOwnerType
+	}
+	if params.Owner.Type != orgOwnerType && params.Owner.Type != "account" {
+		return nil, fmt.Errorf("only owner.type values allowed to create a context are \"organization\" or \"account\", received: %s", params.Owner.Type)
+	}
+	if params.Owner.Id == nil && params.Owner.Slug != nil && params.Owner.Type == "account" {
+		return nil, fmt.Errorf("when creating a context, owner.type with value \"account\" is only allowed when using owner.id and not when using owner.slug")
+	}
+
+	u, err := c.BaseURL.Parse("context")
+	if err != nil {
+		return nil, err
+	}
+	req, err := c.NewRequest("POST", u, &params)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	bodyBytes, err := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		var dest api.ErrorWithMessage
+		if err := json.Unmarshal(bodyBytes, &dest); err != nil {
+			return nil, err
+		}
+		return nil, errors.New(*dest.Message)
+	}
+	var context Context
+	if err := json.Unmarshal(bodyBytes, &context); err != nil {
+		return nil, err
+	}
+
+	return &context, nil
+}
+
+func DeleteContextWithRest(client *rest.Client, contextID string) error {
+	u, err := client.BaseURL.Parse(fmt.Sprintf("context/%s", contextID))
+	if err != nil {
+		return err
+	}
+	req, err := client.NewRequest("DELETE", u, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	bodyBytes, err := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 200 {
+		var dest api.ErrorWithMessage
+		if err := json.Unmarshal(bodyBytes, &dest); err != nil {
+			return err
+		}
+		return errors.New(*dest.Message)
+	}
+
+	return nil
+}
+
+type ListEnvVarsWithRestParams struct {
+	ContextID string
+	PageToken *string
+}
+
+type listEnvVarsResponse struct {
+	Items         []EnvironmentVariable `json:"items"`
+	NextPageToken *string               `json:"next_page_token"`
+}
```

Comment:
I think even your page tokens could be defined as `string`. In many cases, checking against `""` works just as well as checking against `nil`, with less complexity.